UndertaleModLib/Compiler/Compiler.cs

```diff
@@ -54,6 +56,27 @@ public void OnSuccessfulFinish()
                 {
                     foreach (KeyValuePair<string, string> v in GlobalVars)
                         Data?.Variables?.EnsureDefined(v.Key, UndertaleInstruction.InstanceType.Global, false, Data.Strings, Data);
+                    if (Data is not null)
+                    {
+                        foreach (string name in FunctionsToObliterate)
+                        {
+                            string scriptName = "gml_Script_" + name;
+                            UndertaleScript scriptObj = Data.Scripts.ByName(scriptName);
+                            if (scriptObj is not null)
+                                Data.Scripts.Remove(scriptObj);
+                            UndertaleCode codeObj = Data.Code.ByName(scriptName);
+                            if (codeObj is not null)
+                            {
+                                Data.Code.Remove(codeObj);
+                                OriginalCode.ChildEntries.Remove(codeObj);
+                            }
+                            UndertaleFunction functionObj = Data.Functions.ByName(scriptName);
+                            if (functionObj is not null)
+                                Data.Functions.Remove(functionObj);
```

Comment:
Are these instances of `Data.X.Remove()` safe?

Reply:
Should be if you're just normally adding and removing struct definitions, as each struct definition gets its own script, the compiler currently won't let you create `___struct___` functions and I'm fairly sure code, script and function references are all passed by reference instead of ID. But I'm not sure.
(If we don't remove unused struct functions, then adding then removing a struct definition would cause an `Unable to find function` error when launching the game.)